RXSwiftTest/Networking/APIHandler.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  APIHandler.swift
+//  RXSwiftTest
+//
+//  Created by Sunith B on 30/08/19.
+//  Copyright © 2019 Sunith B. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import ReactiveSwift
+
+class APIHandler {
+    
+    //MARK: - Class Members
+    static let shared = APIHandler()
+    
+    //MARK: - Instance Members
+    private var user = User.shared.getCurrentUser()
+    
+    private init() {}
+    
+    //MARK: - Instance Methods
+    func login() -> SignalProducer<Data,RXTestError> {
+        
+        return SignalProducer { [weak self] (observer, lifetime) in
+            guard let `self` = self else { return }
+            Network.shared.makeRequest(request: Endpoint.login()!).observe(on: UIScheduler()).on(event: {(event) in
+                switch event {
+                case .value(let value):
+                    do {
+                        let resultDict = try JSONSerialization.jsonObject(with: value, options: .allowFragments) as! [String:AnyObject]
+                        self.user.updateUser(dict: resultDict)
+                    } catch {
+                        observer.send(error: RXTestError.genericError)
+                    }
+                    break
+                case .failed(let error):
+                    observer.send(error: error)
+                default:
+                    observer.send(error: RXTestError.genericError)
+                }
+            }).flatMap(.latest, { (value) -> SignalProducer<Data, RXTestError> in
+                return self.getBalance()
+            }).flatMap(.latest, { (value) -> SignalProducer<Data, RXTestError> in
+                return self.getTransactions()
+            }).start()
+        }
+        
+    }
+    
+    func getTransactions() -> SignalProducer<Data, RXTestError> {
+        
+        return SignalProducer { [weak self] (observer, lifetime) in
+            Network.shared.makeRequest(request: Endpoint.transactions()!).observe(on: UIScheduler()).on(event: {(event) in
+                switch event {
+                case .value(let value):
+                    guard let transactions = try? JSONSerialization.jsonObject(with: value, options: .allowFragments) as? [[String:Any]] else {return}
```

Comment:
Create an convenience init in Transaction class to avoid parsing here. 

Also return does not throw `observer.send(error: error)`